### ISSUE TITLE
Hcaptch is used in the verification code. How to solve it

### ISSUE THREAD

**Issue by PHslw** [Hecate2/Ignareo-ISML-auto-voter#28]

**Comment by Hecate2** [comment]
Yes I know that but I have not yet written any code for it. You may search for hcaptcha in GitHub and you would see production-ready codes to solve them. So hcaptcha is much much much easier than the original ISML captchas.

**Comment by PHslw** [comment]
I also write code. I'm interested in the verification of this website. When you enter voiting, you will get voiting_ token. Browser post a canvas fingerprint to the server will get a cookie: CI_ Session. When you submit, you will bring a cookie: CI_ Session and voting_ Token and h-captcha-response. I wonder when they were linked? When did IP and device bind?

**Comment by PHslw** [comment]
> Yes I know that but I have not yet written any code for it. I write my codes for my personal interest (telling people that saimoes are just bootcamps for programmers) instead of actually vote for Chtholly. You may search for hcaptcha in GitHub and you would see production-ready codes to solve them. So hcaptcha is much much much easier than the original ISML captchas.

I also write code. I'm interested in the verification of this website. When you enter voiting, you will get voiting_ token. Browser post a canvas fingerprint to the server will get a cookie: CI_ Session. When you submit, you will bring a cookie: CI_ Session and voting_ Token and h-captcha-response. I wonder when they were linked? When did IP and device bind?

**Comment by Hecate2** [comment]
IP is checked by ISML server, and you cannot really detect any js or cookie about your IP. 
The information about your devices is included in the canvas figerprint. If you use the same browser and same machine, the fingerprint would surely be the same. That is why we randomly modify fingerprints.
And maybe you can solve hcaptcha when waiting for submission (instead of solving it after the submitting POST request). I have not tried it yet.

**Comment by PHslw** [comment]
One of my conjectures is that a ticket contains IP and canvas fingerprints, CI_ Session promised that they would put in the same ticket. When an IP or a canvas fingerprint is found in multiple tickets during ticket check-in, it is a duplicate ticket.
Thank you for your reply

**Comment by Hecate2** [comment]
> 

I agree; and I am also Chinese
